Sources/BrowserServicesKit/Statistics/UsageSegmentation.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  UsageSegmentation.swift
+//
+//  Copyright © 2025 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Common
+
+public enum UsageActivityType: String {
+
+    case search
+    case appUse = "app_use"
+
+}
+
+public protocol UsageSegmenting {
+
+    func processATB(_ atb: Atb, withInstallAtb installAtb: Atb, andActivityType activityType: UsageActivityType)
+
+}
+
+public enum UsageSegmentationPixel {
+    case usageSegments
+}
+
+public final class UsageSegmentation: UsageSegmenting {
+
+    public var pixelEvents: EventMapping<UsageSegmentationPixel>?
```

Comment:
PixelFiring replaced with EventMapping to fire pixels in clients as discussed in the tech design